File: Server/src/main/core/game/node/entity/player/link/IronmanMode.java
package core.game.node.entity.player.link;

/**
 * A type of iron man mode.
 *
 * @author Vexia
 */
public enum IronmanMode {
    /**
     * The None.
     */
// HARDCORE_DEAD has to be before Ultimate so that it does not adopt it's restrictions (on the basis of >= in IronmanManager.java?)
	NONE(-1),
    /**
     * Standard ironman mode.
     */
    STANDARD(5),
    /**
     * Hardcore ironman mode.
     */
    HARDCORE(6),
    /**
     * Ultimate ironman mode.
     */
    ULTIMATE(7);

	/**
	 * The icon id.
	 */
	private final int icon;

	/**
	 * Constructs a new {@code IronmanMode} {@code Object}
	 * @param icon the icon.
	 */
	private IronmanMode(int icon) {
		this.icon = icon;
	}

    /**
     * Gets the icon.
     *
     * @return the icon
     */
    public int getIcon() {
		return icon;
	}

}

File: Server/src/main/core/game/node/entity/player/link/IronmanMode.kt
package core.game.node.entity.player.link

/**
 * Ironman mode
 *
 * @property icon Represents the icon associated with the Ironman mode.
 * @constructor Represents the Ironman mode with a specific icon.
 */
enum class IronmanMode(@JvmField val icon: Int) {
    /**
     * None
     *
     * @constructor Represents the absence of any Ironman mode.
     */
    NONE(-1), // Indicates no Ironman mode is selected.

    /**
     * Standard
     *
     * @constructor Represents the standard Ironman mode.
     */
    STANDARD(5), // Represents the standard Ironman mode with a specific icon value.

    /**
     * Hardcore
     *
     * @constructor Represents the hardcore Ironman mode.
     */
    HARDCORE(6), // Represents the hardcore Ironman mode with a specific icon value.

    /**
     * Ultimate
     *
     * @constructor Represents the ultimate Ironman mode.
     */
    ULTIMATE(7) // Represents the ultimate Ironman mode with a specific icon value.
}
